coin_d4_driver/include/coin_d4_driver/lidar_sdk/locker.hpp

```diff
@@ -0,0 +1,370 @@
+// Copyright 2025 ROBOTIS CO., LTD.
+// Authors: Hyeongjun Jeon
+
+#pragma once
+#ifdef _WIN32
+#include <conio.h>
+#include <windows.h>
+#include <process.h>
+#include <tlhelp32.h>
+#include <sys/utime.h>
+#include <io.h>
+#include <direct.h>
+#else
+#include <assert.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <errno.h>
+#endif
+
+class Locker
+{
+public:
+  enum LOCK_STATUS
+  {
+    LOCK_OK = 0,
+    LOCK_TIMEOUT = -1,
+    LOCK_FAILED = -2
+  };
+
+  Locker()
+  {
+#ifdef _WIN32
+    _lock = NULL;
+#endif
+    init();
+  }
+
+  ~Locker()
+  {
+    release();
+  }
+
+  Locker::LOCK_STATUS lock(uint64_t timeout = 0xFFFFFFFF)
+  {
+#ifdef _WIN32
+
+    switch (WaitForSingleObject(
+        _lock,
+        timeout == 0xFFFFFFF ? INFINITE : (DWORD)timeout))
+    {
+      case WAIT_ABANDONED:
+        return LOCK_FAILED;
+
+      case WAIT_OBJECT_0:
+        return LOCK_OK;
+
+      case WAIT_TIMEOUT:
+        return LOCK_TIMEOUT;
+    }
+
+#else
+#ifdef _MACOS
+
+    if (timeout != 0) {
+      if (pthread_mutex_lock(&_lock) == 0) {
+        return LOCK_OK;
+      }
+    }
+
+#else
+
+    if (timeout == 0xFFFFFFFF) {
+      if (pthread_mutex_lock(&_lock) == 0) {
+        return LOCK_OK;
+      }
+    }
+
+#endif
+    else if (timeout == 0) {
+      if (pthread_mutex_trylock(&_lock) == 0) {
+        return LOCK_OK;
+      }
+    }
```

Comment:
There is a cpplint error. 
`If an else has a brace on one side, it should have it on both.`
Is it okay to change the code like this?"
```suggestion
#ifdef _MACOS

    if (timeout != 0) {
      if (pthread_mutex_lock(&_lock) == 0) {
        return LOCK_OK;
      }
    } else if (timeout == 0) {
      if (pthread_mutex_trylock(&_lock) == 0) {
        return LOCK_OK;
      }
    }
#else

    if (timeout == 0xFFFFFFFF) {
      if (pthread_mutex_lock(&_lock) == 0) {
        return LOCK_OK;
      }
    } else if (timeout == 0) {
      if (pthread_mutex_trylock(&_lock) == 0) {
        return LOCK_OK;
      }
    }
#endif
```

or make local boolean variable
```suggestion
    bool tmp = false;
#ifdef _MACOS
    tmp = (timeout != 0);
#else
    tmp = (timeout == 0xFFFFFFFF);
#endif

    if (tmp) {
      if (pthread_mutex_lock(&_lock) == 0) {
        return LOCK_OK;
      }
    } else if (timeout == 0) {
      if (pthread_mutex_trylock(&_lock) == 0) {
        return LOCK_OK;
      }
    }
```

Reply:
I have removed the macro definitions for Windows and macOS. Please review the changes.
https://github.com/ROBOTIS-GIT/lds_03_driver/pull/1/commits/648fd9c11ab4b9c094e1a57fca0808792c97e9c5